src/plugins/Utils.ts

```diff
@@ -234,6 +241,28 @@ class Utils {
 
         return monitoringData;
     }
+
+    static addApplicationTags(applicationTags: any): void {
```

Comment:
I think, it is better to get application tags once and reuse them when needed instead of traversing, processing and checking environment variables every time at each invocation

Reply:
Yes, you are right. I added `ApplicationSupport` class and it parses tags only agent startup.